### DISCUSSION TITLE
go1.22

### DISCUSSION THREAD

**Discussion by ddkwork** [cogentcore/core#869] [Ideas]
Please upgrade to GO1.22 and remove this code

![{E549B0D6-51DC-4432-97F0-2D3737FAC728}](https://github.com/cogentcore/core/assets/19886504/237a4f98-3964-4f2d-b284-c26251c8472f)

**Comment by kkoreilly** [comment]
Indeed, we are planning to do so after 1.22.1 so that any bugs with the new version are fixed first.

**Comment by ddkwork** [comment]
> Indeed, we are planning to do so after 1.22.1 so that any bugs with the new version are fixed first.

I upgraded go1.22 and tested and found no problem https://github.com/ddkwork/core/commit/12bcd56ae0ac61d0c6d98a90fcad50407ecfd930

**Comment by kkoreilly** [comment]
They just released v1.22.1, so I will update Cogent Core to Go 1.22 now.

**Comment by ddkwork** [comment]
I have a way to scan the code to scan all the locations of loopvar, would you be interested in executing it? Or I'll give you a scan result directly, because it's too tiring to look for them one by one



---Original---
From: ***@***.***&gt;
Date: Wed, Mar 6, 2024 23:58 PM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [cogentcore/core] go1.22 (Discussion #869)




 
They just released v1.22.1, so I will update Cogent Core to Go 1.22 now.

**Reply by kkoreilly** [reply]
Yes, after I update it to 1.22 now, I would appreciate a PR from you that gets rid of all of the loopvar usage.

**Reply by kkoreilly** [reply]
I updated it to Go 1.22, so you can sync your fork and then do a PR that gets rid of the loopvar usage.

**Comment by ddkwork** [comment]
I changed it once before, and then the synchronization repository was deleted again, and there are a few places that are not very confirmed, and it may be misoperated, so I'll send you a test to change it



---Original---
From: ***@***.***&gt;
Date: Thu, Mar 7, 2024 01:58 AM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [cogentcore/core] go1.22 (Discussion #869)




 
I updated it to Go 1.22, so you can sync your fork and then do a PR that gets rid of the loopvar usage.

**Comment by ddkwork** [comment]
```go

C:\Users\Admin\Desktop\ux\core>go vet -vettool=C:\Users\Admin\go\bin\copyloopvar.exe ./...
# cogentcore.org/core/core/mobile
# [cogentcore.org/core/core/mobile]
core\mobile\build_test.go:272:5: The copy of the 'for' variable "tc" can be deleted (Go 1.22+)
# cogentcore.org/core/pi/parse
pi\parse\rule.go:1635:4: The copy of the 'for' variable "false" can be deleted (Go 1.22+)
# cogentcore.org/core/gi
# [cogentcore.org/core/gi]
gi\chooser.go:506:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
gi\menu.go:187:4: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
gi\menu.go:188:4: The copy of the 'for' variable "s" can be deleted (Go 1.22+)
gi\widget.go:493:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
gi\widget.go:511:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
gi\label_test.go:37:4: The copy of the 'for' variable "st" can be deleted (Go 1.22+)
gi\layout_test.go:68:3: The copy of the 'for' variable "wrap" can be deleted (Go 1.22+)
gi\layout_test.go:70:4: The copy of the 'for' variable "dir" can be deleted (Go 1.22+)
gi\layout_test.go:72:5: The copy of the 'for' variable "align" can be deleted (Go 1.22+)
gi\layout_test.go:99:3: The copy of the 'for' variable "wrap" can be deleted (Go 1.22+)
gi\layout_test.go:105:4: The copy of the 'for' variable "dir" can be deleted (Go 1.22+)
gi\layout_test.go:107:5: The copy of the 'for' variable "align" can be deleted (Go 1.22+)
gi\layout_test.go:133:3: The copy of the 'for' variable "align" can be deleted (Go 1.22+)
gi\layout_test.go:152:3: The copy of the 'for' variable "ai" can be deleted (Go 1.22+)
gi\layout_test.go:153:3: The copy of the 'for' variable "align" can be deleted (Go 1.22+)
gi\layout_test.go:176:3: The copy of the 'for' variable "ai" can be deleted (Go 1.22+)
gi\layout_test.go:177:3: The copy of the 'for' variable "align" can be deleted (Go 1.22+)
gi\layout_test.go:418:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
gi\layout_test.go:419:3: The copy of the 'for' variable "sz" can be deleted (Go 1.22+)
gi\snackbar_test.go:80:3: The copy of the 'for' variable "tm" can be deleted (Go 1.22+)
gi\snackbar_test.go:95:3: The copy of the 'for' variable "tm" can be deleted (Go 1.22+)
# cogentcore.org/core/giv
# [cogentcore.org/core/giv]
giv\fileview.go:287:4: The copy of the 'for' variable "sp" can be deleted (Go 1.22+)
giv\settingsview.go:57:3: The copy of the 'for' variable "se" can be deleted (Go 1.22+)
giv\sliceview.go:727:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
giv\sliceview.go:728:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
giv\sliceview.go:830:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
giv\sliceviewinline.go:173:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
giv\structview.go:290:3: The copy of the 'for' variable "vv" can be deleted (Go 1.22+)
giv\structviewinline.go:117:3: The copy of the 'for' variable "vv" can be deleted (Go 1.22+)
giv\structviewinline.go:118:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
giv\tableview.go:274:3: The copy of the 'for' variable "fli" can be deleted (Go 1.22+)
giv\tableview.go:355:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
giv\tableview.go:356:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
giv\tableview.go:386:4: The copy of the 'for' variable "fli" can be deleted (Go 1.22+)
giv\tableview.go:477:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
giv\tableview.go:503:4: The copy of the 'for' variable "fli" can be deleted (Go 1.22+)
giv\timeview.go:285:3: The copy of the 'for' variable "yd" can be deleted (Go 1.22+)
# cogentcore.org/core/greasi
greasi\gui.go:23:4: The copy of the 'for' variable "cmd" can be deleted (Go 1.22+)
# cogentcore.org/core/examples/layout
# [cogentcore.org/core/examples/layout]
examples\layout\layout.go:111:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)
examples\layout\layout.go:112:3: The copy of the 'for' variable "sz" can be deleted (Go 1.22+)
# cogentcore.org/core/xyzv
xyzv\sceneview.go:148:3: The copy of the 'for' variable "i" can be deleted (Go 1.22+)

C:\Users\Admin\Desktop\ux\core>  

```

**Reply by kkoreilly** [reply]
Do you want to get rid of those unnecessary loopvar copies in a PR, or do you want me to do it?

**Comment by ddkwork** [comment]
yo do,Because there's a couple of places I'm not sure about.




2762713521
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人:                                                                                                                        "cogentcore/core"                                                                                    ***@***.***&gt;;
发送时间:&nbsp;2024年3月7日(星期四) 中午11:21
***@***.***&gt;;
***@***.***&gt;;"State ***@***.***&gt;;
主题:&nbsp;Re: [cogentcore/core] go1.22 (Discussion #869)





 
Do you want to get rid of those unnecessary loopvar copies in a PR, or do you want me to do it?

**Reply by kkoreilly** [reply]
I did it in c812a052cea66ca1e8afd34c54065a4ef3df4ec0. There were some warnings that were in the wrong place, and one that was not correct, but overall it was good. Thanks for giving me the vet output!

**Comment by ddkwork** [comment]
ok



---Original---
From: ***@***.***&gt;
Date: Thu, Mar 7, 2024 11:39 AM
To: ***@***.***&gt;;
Cc: ***@***.***&gt;;"State ***@***.***&gt;;
Subject: Re: [cogentcore/core] go1.22 (Discussion #869)




 
I did it in c812a05. There were some warnings that were in the wrong place, and one that was not correct, but overall it was good.
